### PR TITLE
docs: update subcommand plugin guide

### PR DESCRIPTION
**Pull request by feefs**
This updates the subcommand plugin guide now that application commands are supported by the plugin.

### CODE REVIEW

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 1]
```diff
@@ -25,21 +28,20 @@ export class UserCommand extends Subcommand {
...
       ...options,
+      name: 'vip',
       subcommands: [
         {
-          name: 'add',
-          messageRun: 'messageAdd',
-          chatInputRun: 'chatInputAdd'
+          name: 'list',
+          messageRun: 'messageList',
+          default: true // Only functional for message subcommands
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:36]: This comment should be added outside a code example, in the guide!

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 2]
```diff
@@ -50,19 +52,128 @@ export class UserCommand extends Subcommand {
   public async messageRemove(message: Message, args: Args) {}
 
   public async messageList(message: Message, args: Args) {}
+}
+```
 
-  public async chatInputAdd(interaction: Subcommand.ChatInputInteraction) {}
+In this example command, "vip list" will run `messageList`, "vip add" will run `messageAdd`, and "vip remove" will run
+`messageRemove`. You can map the value of `messageRun` to the name of any method in the class you want, but we recommend
+the naming convention above to keep things organized.
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:60]:
```suggestion
`messageRemove`. You can map the value of `messageRun` to any name of a method in the class, but we recommend you use a
naming convention to keep things organized (personally we recommend prefixing the method name with the type it's for (`messageAdd` for messages and `chatInputAdd` for chat input)).
```

File: docs/Guide/plugins/Subcommands/getting-started.mdx [hunk 3]
```diff
@@ -50,19 +52,128 @@ export class UserCommand extends Subcommand {
...
+  public constructor(context: Subcommand.Context, options: Subcommand.Options) {
+    super(context, {
+      ...options,
+      name: 'vip',
+      subcommands: [
+        {
+          name: 'list',
+          messageRun: 'messageList',
+          chatInputRun: 'chatInputList',
+          default: true // Only functional for message subcommands
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/getting-started.mdx:157]: Ditto

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 4]
```diff
@@ -1,49 +1,27 @@
 ---
-title: Using Subcommand groups
+title: Using Subcommand Groups for Chat Input Commands
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:2]: But message commands can use them too!

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 5]
```diff
@@ -1,49 +1,27 @@
...
 sidebar_position: 2
 ---
 
-Before we start, we assume you already know what subcommand groups are, if not please refer to the
-[Official Discord Docs](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups)
-for more information.
+The following example will build off the command in the previous section. As a disclaimer, the result on the user side
+will seem redundant when compared to the previous command. Regardless, the purpose is to show how to use Subcommand
+Groups, rather than a useful example.
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:8]:
```suggestion
The following examples shows off how you can implement support for mapping subcommands that are part of a subcommand group
to a method in your command class. This example does build off of the previous section, so if you jumped here
directly without checking it out, we suggest you go back there, take a peek, and come back fresh and ready to code!
```

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 6]
```diff
@@ -1,49 +1,27 @@
...
 
-## Message Subcommand Groups
+Since Subcommand Groups are a little nuanced, please refer to Discord's
+[Subcommands and Subcommand Groups](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups)
+documentation to see better examples of applying them.
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:12]:
```suggestion
Subcommand groups are a way to...well...group subcommands by some criteria you want (for instance all subcommands affecting users can go into one `user` group). Discord's
[Subcommands and Subcommand Groups](https://discord.com/developers/docs/interactions/application-commands#subcommands-and-subcommand-groups) explanation might give you some better examples for how you can use them for your users' benefits! Here, we are just showing you how to implement the mappings for them using the Subcommands plugin! 
```

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx [hunk 7]
```diff
@@ -1,49 +1,27 @@
...
-  public async messageAdd(message: Message, args: Args) {}
-
-  public async messageRemove(message: Message, args: Args) {}
-
-  public async messageList(message: Message, args: Args) {}
-}
-```
+To use subcommands with Chat Input Commands (Slash Commands), we recommend you learn how to register and use regular
+Chat Input Commands first in order to understand `registerApplicationCommands()`. Please read [Registering Chat Input
+Commands][register] for details.
```
Comment (vladfrangu) [inline docs/Guide/plugins/Subcommands/subcommand-groups.mdx:20]:
```suggestion
Before you can use subcommands with Chat Input Commands (Slash Commands), we recommend you learn how to register and use regular
Chat Input Commands first in order to understand what the `registerApplicationCommands()` method does. Please read [Registering Chat Input
Commands][register] for details, and once you're done, come back here!
```

**Review by vladfrangu — CHANGES_REQUESTED** [review]

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx (hunk 4 above)
Comment (feefs) [reply docs/Guide/plugins/Subcommands/subcommand-groups.mdx]: Added a message command section.

File: docs/Guide/plugins/Subcommands/subcommand-groups.mdx (hunk 6 above)
Comment (feefs) [reply docs/Guide/plugins/Subcommands/subcommand-groups.mdx]: I rewrote this when adding the message commands section, let me know if there are other changes needed.

**Review by vladfrangu — APPROVED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**